gen_bt_data.py

```diff
@@ -39,31 +41,147 @@ def linearModel(startDate, endDate, startPrice, endPrice, deltaTime, spread, dig
+        }]
+        i += 1
+        timestamp += deltaTime
+        bidPrice += lift
+        askPrice = bidPrice + spread
+        (bidVolume, askVolume) = volumesFromTimestamp(timestamp, spread)
+    return ticks
+
+
+def waveModel(startDate, endDate, startPrice, endPrice, deltaTime, spread, volatility):
+    timestamp = startDate
+    bidPrice = startPrice
+    askPrice = bidPrice + spread
+    bidVolume = 1
+    askVolume = bidVolume + spread
+    deltaPrice = endPrice - startPrice
+    count = ceil((endDate + datetime.timedelta(days=1) - startDate)/deltaTime)
+    d = count/2  # Denominator for curve shaping
+    ticks = []
+    for i in range(0, count):
+        ticks += [{
+            'timestamp': timestamp,
+             'bidPrice': bidPrice,
+             'askPrice': askPrice,
+            'bidVolume': bidVolume,
+            'askVolume': askVolume
+        }]
+        i += 1
+        timestamp += deltaTime
+        # Select appropriate formula depending on starting and ending prices
+        if abs(deltaPrice) > 0:
+            bidPrice = abs(startPrice + i/(count - 1)*deltaPrice + volatility*sin(i/(count - 1)*3*pi))
+        else:
+            bidPrice = abs(startPrice + (volatility*sin(i/(count - 1)*3*pi)))
+        askPrice = bidPrice + spread
+        (bidVolume, askVolume) = volumesFromTimestamp(timestamp, spread)
+    return ticks
+
+
+def curveModel(startDate, endDate, startPrice, endPrice, deltaTime, spread, volatility):
+    timestamp = startDate
+    bidPrice = startPrice
+    askPrice = bidPrice + spread
+    bidVolume = 1
+    askVolume = bidVolume + spread
+    deltaPrice = endPrice - startPrice
+    count = ceil((endDate + datetime.timedelta(days=1) - startDate)/deltaTime)
+    d = count/volatility  # A kind of volatility interpretation via curve shaping
+    ticks = []
+    for i in range(0, count):
+        ticks += [{
+            'timestamp': timestamp,
+             'bidPrice': bidPrice,
+             'askPrice': askPrice,
+            'bidVolume': bidVolume,
+            'askVolume': askVolume
+        }]
+        i += 1
+        timestamp += deltaTime
+        bidPrice = startPrice + (1 - (exp(i/d) - exp((count - 1)/d))/(1 - exp((count - 1)/d)))*deltaPrice
+        askPrice = bidPrice + spread
+        (bidVolume, askVolume) = volumesFromTimestamp(timestamp, spread)
+    return ticks
+
+
+def randomModel(startDate, endDate, startPrice, endPrice, deltaTime, spread, volatility):
+    timestamp = startDate
+    bidPrice = startPrice
+    askPrice = bidPrice + spread
+    bidVolume = 1
+    askVolume = bidVolume + spread
+    ticks = []
+    while timestamp < (endDate + datetime.timedelta(days=1)):
+        ticks += [{
+            'timestamp': timestamp,
+             'bidPrice': bidPrice,
+             'askPrice': askPrice,
+            'bidVolume': bidVolume,
+            'askVolume': askVolume
+        }]
+        timestamp += deltaTime
+        bidPrice = 1 + volatility*random.random()
```

Comment:
Can we have relative price based on the previous value? For example given `start` & `end` value, 1 `volatility` could mean max 1% of change between `min` & `max`, however each random will be towards `end` value (e.g. `curr_value = prev_value + random + linear_avg_price` (where `random` is no more than 1% of `max-min`).